core/lib/spree/core/search/variant.rb

```diff
@@ -0,0 +1,43 @@
+module Spree
+  module Core
+    module Search
+      class Variant < Base
+        def initialize(params)
+          super
+          params[:scope] ||= Spree::Variant.all
+          params[:q] ||= ''
+        end
+
+        def search
+          # Syntax inferred from http://ransack-demo.herokuapp.com/users/advanced_search
+          query = { g: {} }
+          grouping_count = 0
+          fields = ['option_values_name', 'product_name', 'sku']
+          params[:q].split(' ').each do |part|
+            query[:g][grouping_count.to_s] = {
+              m: 'or',
+              c: {}
+            }
+            condition_count = 0
+            fields.each do |field|
+              query[:g][grouping_count.to_s][:c][condition_count.to_s] = {
+                a: { '0' => { name: field } },
+                p: 'cont',
+                v: { '0' => { value: part } }
+              }
+              condition_count += 1
+            end
+            grouping_count += 1
+          end
+
+          result = params[:scope]
+            .ransack(query)
```

Comment:
Place the . on the previous line, together with the method call receiver.<br>Align the operands of an expression in an assignment spanning multiple lines.